Explorer/Assets/Scripts/Global/Dynamic/DynamicECSWorldExtensions.cs

```diff
@@ -0,0 +1,22 @@
+using Arch.Core;
+using DCL.Character.Components;
+using DCL.Profiles;
+
+namespace Global.Dynamic
+{
+    public static class DynamicECSWorldExtensions
+    {
+        private static readonly Entity[] OWN_PROFILE_BUFFER = new Entity[1];
+
+        public static void SetProfileToOwnPlayer(this World world, Profile profile)
```

Comment:
You don't need to query the player from ECS: this logic is executed after `GlobalWorld Create` so you have `playerEntity` already: just return it from the factory

Reply:
I see two issues by using the `playerEntity`:
1. We are not able to check if the entity is valid, so we cannot delete or change at any point the entity of the player, which may bring some limitations.
2. We gotta find workarounds to get access to the `playerEntity` by passing references originated at `GlobalWorldFactory`

Do you think this query will have a significant performance impact? Because its only called once at the moment. The world should have this component indexed, right?

Reply:
I think we keep one player entity and never destroy it by design. Otherwise, your query won't be helpful either if for example there is no player entity at all.
Moreover in systems we use `playerEntity` directly already because it's a shared persistent entity by design.